website/server/libs/spells.js

```diff
@@ -163,14 +163,32 @@ async function castSpell (req, res, { isV3 = false }) {
       task: results[1],
     });
   } else if (targetType === 'self') {
-    await castSelfSpell(req, user, spell, quantity);
-
-    let userToJson = user;
-    if (isV3) userToJson = await userToJson.toJSONWithInbox();
-
-    res.respond(200, {
-      user: userToJson,
-    });
+    // Check if chilling frost or stealh skill has been previously casted or not.
+    // See #12361 for more details.
+    const spellName = spell.key;
+    const incompleteDailiesDue = await Tasks.Task.find({
+      userId: user._id,
+      type: 'daily',
+      $and: [
+        {type: 'daily', completed: false, isDue: true },
+      ],
+    }).exec();
+    if (spellName === 'frost' && user.stats.buffs.streaks) {
```

Comment:
for `frost` since we only need  `stats.buffs.streaks` to see if it can be used or not and that is available in common code as well could we move this check there? so that we don't have to add a new one on the client

Reply:
I have added the check in the `common` code. Please have a look at it if I need to change something.